ice-phishing/src/agent.js

```diff
@@ -469,46 +593,70 @@ const provideHandleTransaction = (provider) => async (txEvent) => {
+          counters.totalERC721Approvals += 1;
+          counters.detectedERC721ApprovalsInfo += 1;
+          const anomalyScore = counters.detectedERC721ApprovalsInfo / counters.totalERC721Approvals;
+          findings.push(createHighNumApprovalsInfoAlertERC721(spender, approvalsInfoSeverity[spender], anomalyScore));
         }
 
         if (isApprovalForAll) {
           if (
             approvalsForAll721InfoSeverity[spender] &&
             approvalsForAll721InfoSeverity[spender].length > approveForAllCountThreshold
           ) {
-            findings.push(createApprovalForAllInfoAlertERC721(spender, owner, asset));
+            counters.totalERC721ApprovalsForAll += 1;
+            counters.detectedERC721ApprovalsForAllInfo += 1;
+            const anomalyScore = counters.detectedERC721ApprovalsForAllInfo / counters.totalERC721ApprovalsForAll;
+            findings.push(createApprovalForAllInfoAlertERC721(spender, owner, asset, anomalyScore, hash));
           } else if (
             approvalsForAll1155InfoSeverity[spender] &&
             approvalsForAll1155InfoSeverity[spender].length > approveForAllCountThreshold
           ) {
-            findings.push(createApprovalForAllInfoAlertERC1155(spender, owner, asset));
+            counters.totalERC1155ApprovalsForAll += 1;
+            counters.detectedERC1155ApprovalsForAllInfo += 1;
+            const anomalyScore = counters.detectedERC1155ApprovalsForAllInfo / counters.totalERC1155ApprovalsForAll;
+            findings.push(createApprovalForAllInfoAlertERC1155(spender, owner, asset, anomalyScore, hash));
           }
         }
       } else {
         if (approvalsERC20[spender] && approvalsERC20[spender].length > approveCountThreshold) {
-          findings.push(createHighNumApprovalsAlertERC20(spender, approvals[spender]));
+          counters.totalERC20Approvals += 1;
+          counters.detectedERC20Approvals += 1;
+          const anomalyScore = counters.detectedERC20Approvals / counters.totalERC20Approvals;
+          findings.push(createHighNumApprovalsAlertERC20(spender, approvals[spender], anomalyScore));
         }
 
         if (approvalsERC721[spender] && approvalsERC721[spender].length > approveCountThreshold) {
-          findings.push(createHighNumApprovalsAlertERC721(spender, approvals[spender]));
+          counters.totalERC721Approvals += 1;
+          counters.detectedERC721Approvals += 1;
+          const anomalyScore = counters.detectedERC721Approvals / counters.totalERC721Approvals;
+          findings.push(createHighNumApprovalsAlertERC721(spender, approvals[spender], anomalyScore));
         }
 
         if (isApprovalForAll) {
           if (approvalsForAll721[spender] && approvalsForAll721[spender].length > approveForAllCountThreshold) {
-            findings.push(createApprovalForAllAlertERC721(spender, owner, asset));
+            counters.totalERC721ApprovalsForAll += 1;
+            counters.detectedERC721ApprovalsForAll += 1;
```

Comment:
```suggestion
            counters.totalERC721ApprovalsForAll += approvalsForAll721[spender].length;
            counters.detectedERC721ApprovalsForAll += approvalsForAll721[spender].length;
```
This alert is not supposed to be anomalous since it is practically just emitting for every instance. So perhaps one is fine, but updating to use `.length` would use the correct numbers. Thoughts?

Also, L647 should be updated to use `>=` since the threshold is `1` and currently it is missing instances where `.length` is `1`.

Reply:
In regards to the 2nd part of the comment, as in all of these checks `>` is used, should I maybe decrement the threshold in the config? Also, is your suggestion only about the `ApprovalForAll`s or about all thresholds?